File: backend/app/schemas/person_history.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime, date

# Schema สำหรับแสดงผล person_history
class PersonHistoryOut(BaseModel):
    id: int
    person_id: Optional[int]
    personal_id_number: Optional[str]
    first_name: Optional[str]
    middle_name: Optional[str]
    last_name: Optional[str]
    nick_name: Optional[str]
    birth_date: Optional[date]
    gender_type_id: Optional[int]
    marital_status_type_id: Optional[int]
    country_id: Optional[int]
    height: Optional[int]
    weight: Optional[int]
    racial_type_id: Optional[int]
    income_range_id: Optional[int]
    about_me: Optional[str]
    action: Optional[str]
    action_at: datetime
    action_by: Optional[int]

    class Config:
        from_attributes = True

File: frontend/src/services/PersonHistory.ts
import axios from 'axios';
import Cookies from 'js-cookie';

const BASE_URL = 'http://localhost:8080';

interface PersonHistory {
  id: number;
  person_id: number | null;
  personal_id_number: string | null;
  first_name: string | null;
  middle_name: string | null;
  last_name: string | null;
  nick_name: string | null;
  birth_date: string | null;
  gender_type_id: number | null;
  marital_status_type_id: number | null;
  country_id: number | null;
  height: number | null;
  weight: number | null;
  racial_type_id: number | null;
  income_range_id: number | null;
  about_me: string | null;
  action: string | null;
  action_at: string;
  action_by: number | null;
}

export const getPersonHistory = async (): Promise<PersonHistory[]> => {
  const token = Cookies.get('access_token');
  if (!token) {
    throw new Error('No access token found');
  }
  const response = await axios.get(`${BASE_URL}/person_history`, {
    headers: { Authorization: `Bearer ${token}` },
  });
  return response.data;
};